### ISSUE TITLE
support for config from toml file

### ISSUE THREAD

**Issue by prathapbelli** [pypa/pip-audit#644]
if I am not wrong as pip audit uses OSV internally,
osv-scanner supports the config from toml file, it will be easy to maintain if pip audit provides argument to take config from toml to set config like ignoring vulnerable.

**Comment by woodruffw** [comment]
Hi @prathapbelli! In the future, please use our issue templates. In particular, the "Feature Request" template is probably the appropriate one here.

By default, `pip-audit` does not use the OSV REST API -- it uses PyPI's vulnerability API, which is similar but distinct. It _can_ use the OSV API via `-s osv`, but we generally don't encourage users to do that unless they have a specific reason to avoid the PyPI API (e.g., firewall restrictions). 

Given that, it's unlikely that we could directly reuse the same config format the `osv-scanner` uses. Doing so would also pose challenges for integration with `pip`, which is on our eventual roadmap -- `pip` has its own configuration file, so we don't want to either modify it or add another file into the mix without coordinating with them.

TL;DR: This is probably out of scope for us, currently. Our recommendation is to use the CLI flags for things like vulnerability masking; if your use case involves large numbers of ignored vulnerabilities, then you can create a wrapper script that munges them into the `pip-audit` CLI.

**Comment by Kristinita** [comment]
**Type: Question** :question:

> By default, pip-audit does not use the OSV REST API -- it uses PyPI's vulnerability API, which is similar but distinct. It can use the OSV API via `-s osv`, but we generally don't encourage users to do that unless they have a specific reason to avoid the PyPI API (e.g., firewall restrictions).

@woodruffw, do these words mean that users get the same result if they use the option `--vulnerability-service osv`, and they shouldn’t use this option “unless they have a specific reason to avoid the PyPI API (e.g., firewall restrictions)”? Or in which cases users should use the option `--vulnerability-service osv`?

Thanks.

**Comment by woodruffw** [comment]
@Kristinita 99% of users should probably use the default, i.e. the PyPI API -- the only difference between the PyPI and OSV APIs is that the PyPI API pulls from the [PYSEC](https://github.com/pypa/advisory-database) DB, which itself gets populated from OSV.

The reason for this is that PYSEC advisories can generally be updated immediately, meaning we can fix or withdraw incorrect advisories faster than OSV can (which is already a lot faster than CVE).

TL;DR: You'll get the same results 99% of the time, but the PyPI service will occasionally have more timely results. That plus its integration into PyPI's APIs is why we recommend it and use it by default.
